src/main/java/com/github/packageurl/PackageURL.java

```diff
@@ -276,7 +276,9 @@ private String validateNamespace(final String[] values) throws MalformedPackageU
             case StandardTypes.GITHUB:
             case StandardTypes.GOLANG:
             case StandardTypes.RPM:
-                retVal = tempNamespace.toLowerCase();
+                if (tempNamespace != null) {
+                    retVal = tempNamespace.toLowerCase();
```

Comment:
```suggestion
                    retVal = tempNamespace.toLowerCase(Locale.ROOT);
```
In the Turkish locale `I` becomes `ı`.

Reply:
@ppkarwasz Thanks for the review. I am surprised that my IDE does not flag missing `Locale`. Can you provide me a test case with an actual word that might be used and I will add it.

Reply:
@ppkarwasz There are actually nine total occurrences of `toLowerCase()` to this class.

Reply:
So, the `type` itself must be ASCII lowercase, as well as the `qualifiers` keys, but the rest is presumably can be UTF-8. I only see one type in the spec (luarocks) which requires ASCII lowercase.

I wonder if an ASCII-specific `toLowerCase` should be used for type and keys, and for the rest it's sort of ambiguous. It's supposed to be percent-encoded, but I guess this only applies to the PURL canonical output and not the input.

Reply:
I am unconvinced that the lowercase should produce different output on different locales. This actually makes the canonical URL returned different in Turkey than the US, which is probably not what is intended.

I think that any time that these fields are lowercased they are meant to be ASCII even though the spec is ambiguous. Otherwise, it is not "normalized" which is what is intended, I think.

I did find in the code lots of repeated code that could go into a method, and lots of repeated lowercase calls that ideally should only happen once, so this could be fixed.

Reply:
> I think that any time that these fields are lowercased they are meant to be ASCII even though the spec is ambiguous. Otherwise, it is not "normalized" which is what is intended, I think.

That is exactly what I am suggesting: we can not call `toLowerCase()` without a locale parameter, because this would use the default locale. Since these names are namespaces and package names, we should use a fixed locale. `Locale.ROOT` accomplishes this, without being USA-centric.

Reply:
I think it may be preferable to use an ASCII-specific lowercase helper method, then, but I would do this in a separate pull request.

For the fields that need to be lowercased, I think an additional check may need to be performed to verify if the original string in ASCII first.